doc/api/cli.md

```diff
@@ -894,6 +894,18 @@ generated as part of the test runner output. If no tests are run, a coverage
 report is not generated. See the documentation on
 [collecting code coverage from tests][] for more details.
 
+### `--experimental-test-isolation`
+
+<!-- YAML
+added:
+  - REPLACEME
+-->
+
+When running tests with the `node:test` module,
+each test file is run in its own process.
+running `--experimental-test-isolation=none` will run all
+files in the same process.
```

Comment:
Could we maybe make it a feature reserved to `run()`? We can always add the flag later, but it seems to me better if we can avoid adding the flag until we know we want to keep the feature along (and if a boolean flag is what makes sense, as we might want to give option for running tests in worker threads instead).

Reply:
I prefer to keep the flag as well since I believe this feature makes a lot of sense. I have seen a lot of use cases justifying this:
- typescript, which needs to re-compile per each swpaned process
- setup, for example initializing a server that has a penalty for running again and again per process
and many other variations of this tradeoff